Kernel/PreviousMode.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+namespace Kernel {
+
+enum class PreviousMode : u8 {
```

Comment:
I realize this is a refactor but perhaps "ExecutionMode" or "PermissionMode" would be more descriptive? At some point the "PreviousMode" will become the "CurrentMode" and at that point the name is a bit silly.

Reply:
I renamed the enum to ExecutionMode.